src/lib.rs

```diff
@@ -594,6 +595,23 @@ impl HidDevice {
         self.inner.get_feature_report(buf)
     }
 
+    // Send a Output report to the device.
+    //
+    // Output reports are sent over the Control endpoint as a Set_Report
+    // transfer. The first byte of data[] must contain the Report ID.
+    // For devices which only support a single report, this must be set
+    // to 0x0. The remaining bytes contain the report data. Since the
+    // Report ID is mandatory, calls to hid_send_output_report() will
+    // always contain one more byte than the report contains. For example,
+    //  if a hid report is 16 bytes long, 17 bytes must be passed to
+    //  hid_send_output_report(): the Report ID (or 0x0, for devices
+    // which do not use numbered reports), followed by the report
+    // data (16 bytes). In this example, the length passed in
+    // would be 17.
+    pub fn send_output_report(&self, data: &[u8]) -> HidResult<()> {
```

Comment:
Change to:

```rust
#[cfg(hidapi)]
pub fn send_output_report(&self, data: &[u8]) -> HidResult<()> {
```

Reply:
Sorry for that, I didn't realized at first that you support also `windows-native` and `linux-native`.

Maybe I will wait with adding the `#[cfg(hidapi)]`. You already provided `linux-native` implementation and I can try to write `windows-native`, too. I have such device (Redragon k585 keyboard uses `send_output_report` to control RGB leds) and I will be able to test it.

But please be patient, because testing it on Windows will take me more time and I'm also very short of my time :/

Reply:
> Seems like there is not actually a release of https://github.com/libusb/hidapi yet with the feature. Also, your PR only makes the new set_output_report work with the hidapi backend.

I only don't know what to do with that.

Will you be ok to use unreleased version like in my PR?

Or do you see any other option? Like adding new feature (`hidapi_experimental`?). But I don't know what's the best way to conditionally use different commit for git submodule for that feature (additional `git checkout` in `build.rs` or having two git submodules - `hidapi` & `hidapi_experimental`?)....

Reply:
I am totally ok with updating the module on the git master branch of my repo. Not yet sure if I want to make a crates.io release yet, but I probably will. Anyways, its not a problem for this pull request.

Reply:
Thx! So, I will add `linux-native` and `windows-native` implementations for completeness.

Reply:
Nice!